tests/integration/targets/cloud_discovery_providers_aws/tasks/main.yml

```
@@ -0,0 +1,485 @@
+---
+- module_defaults:
+    group/infoblox.universal_ddi.all:
+      portal_url: "{{ portal_url }}"
+      portal_key: "{{ portal_key }}"
+
+  block:
+    # Create a cloud discovery with a random name to avoid conflicts
+    - ansible.builtin.set_fact:
+        cloud_discovery_provider_aws_name: "test_aws_{{ 999999 | random | string }}"
+        cloud_discovery_provider_aws_name_1: "test_aws_{{ 999999 | random | string }}"
+        cloud_discovery_provider_azure_name: "test_azure_{{ 999999 | random | string }}"
+        cloud_discovery_provider_gcp_name: "test_gcp_{{ 999999 | random | string }}"
+        config_access_id: "arn:aws:iam::{{ 999999999999 | random | string }}:role/infoblox_discovery"
+        config_access_id_1: "arn:aws:iam::{{ 999999999999 | random | string }}:role/infoblox_discovery"
+
+
+    - name: Create an AWS cloud discovery provider (check mode)
+      infoblox.universal_ddi.cloud_discovery_providers:
+        name: "{{ cloud_discovery_provider_aws_name }}"
+        provider_type: "Amazon Web Services"
+        account_preference: "single"
+        credential_preference:
+          access_identifier_type: "role_arn"
+          credential_type: "dynamic"
+        source_configs:
+          - credential_config:
+                access_identifier: "{{ config_access_id }}"
+        state: present
+      check_mode: true
+      register: aws_provider
+    - name: Get Infoblox AWS cloud discovery provider
```

```suggestion
    - name: Get AWS cloud discovery provider
```
`nit`: I think Infoblox keyword can be removed from all the task names.

---

done

---

Please make this change for other providers too!